scheduler/system_util.go

```diff
@@ -288,7 +309,78 @@ func diffSystemAllocsForNode(
 			result.place = append(result.place, allocTuple)
 		}
 	}
-	return result
+
+	finalResult := new(diffResult)
+	for _, result := range results {
+		finalResult.Append(result)
+	}
+
+	return finalResult
+}
+
+// ensureMaxSystemAllocCount enforces the invariant that the per-node diffResult
+// we have for a system or sysbatch job should never have more than "count"
+// allocations in a desired-running state.
+func ensureMaxSystemAllocCount(result *diffResult, count int) {
+
+	// sort descending by TaskGroup name, then JobModifyIndex, then
```

Comment:
where do we sort by TaskGroup name?

Reply:
Oops, we don't. I added that in my first attempt and forgot to remove it. Will fix.

Reply:
Fixed in 661fd31